airflow/providers/amazon/aws/auth_manager/aws_auth_manager.py

```diff
@@ -443,6 +443,62 @@ def batch_is_authorized_variable(
         ]
         return self.avp_facade.batch_is_authorized(requests=facade_requests, user=self.get_user())
 
+    def filter_permitted_dag_ids(
+        self,
+        *,
+        dag_ids: set[str],
+        methods: Container[ResourceMethod] | None = None,
+        user=None,
+    ):
+        """
+        Filter readable or writable DAGs for user.
+
+        :param dag_ids: the list of DAG ids
+        :param methods: whether filter readable or writable
+        :param user: the current user
+        """
+        if not methods:
+            methods = ["PUT", "GET"]
```

Comment:
Aren't methods supposed to be of type `Container[ResourceMethod]`? Here you're just using strings

Reply:
Yeah.  In L370 in base_auth_manager (above) they are using string literals as well though.

Reply:
ResourceMethod is a literal defined here: https://github.com/apache/airflow/blob/main/airflow/auth/managers/base_auth_manager.py#L60